In [30]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from mpl_toolkits.mplot3d import Axes3D
from sklearn.linear_model import LinearRegression

In [ ]:
df = pd.read_csv('kadai4.csv')

面積比と、色は以下のように値を変更した。

+ 面積比 1:2, 1:1, 2:1 ----> -1, 0, 1
+ 色 cool:-1, middle:0, warm:1

In [37]:
df

Area ratio  Left  Right  Elegance
0            0     1      1         1
1            0     0      0         2
2           -1    -1     -1         2
3            1     1      1         1
4           -1     0      0         1
5            0    -1     -1         1
6            0     1      0         0
7            0     0      0        -1
8            0     0     -1        -2
9           -1     1      0        -1
10          -1     0      0         0
11          -1     0     -1        -1

In [38]:
y = df[df.columns[-1]]
x = df.drop(df.columns[-1], axis=1)

y = np.array(y)
x = np.array(x)

In [63]:
clf = LinearRegression(normalize=True)
clf.fit(x, y)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=True)

In [75]:
# 回帰係数, 切片 (誤差), 決定係数
coef = clf.coef_
intercept = clf.intercept_
score = clf.score(x, y)
print(coef, intercept, score)

[-0.06756757 -1.87837838  1.90540541] 0.8581081081081079 0.5146242132543504


In [65]:
pd.DataFrame(coef, index=df.columns[0:3], columns=['coef'])

coef
Area ratio -0.067568
Left       -1.878378
Right       1.905405

上記のプログラムより以下の回帰式を求めた

+ 優雅 = -0.067568 * 面積比 + -1.878378 * 左側色 + 1.905405 * 右側色 + 0.8581081081081079
+ R^2 = 0.5146242132543504  
  
回帰式から、優雅に貢献しているデザイン要素は左右の色であることが判明した。それぞれの回帰系数を見ると、左側は負の値で、右側は正の値であることから、左側が寒色で右側が暖色であれば、優雅であると考えられる。 